src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1765,6 +1767,22 @@ private void EmitObjectCreationExpression(BoundObjectCreationExpression expressi
             }
         }
 
+        private bool ConstructorNotSideEffecting(MethodSymbol constructor)
+        {
+            var originalDef = constructor.OriginalDefinition;
+            var compilation = _module.Compilation;
+
+            return originalDef == compilation.GetSpecialTypeMember(SpecialMember.System_Nullable_T__ctor) ||
+                originalDef == compilation.GetWellKnownTypeMember(WellKnownMember.System_ValueTuple_T1__ctor) ||
```

Comment:
There are many checks here that would typically fail. Is it possible to make a cheap easy-out check, like for the name of the type, before doing all the tuple checks?